dom.bs

```diff
@@ -6484,8 +6514,19 @@ string <var>namespace</var> (default null):</p>
 
  <li><p>If <var>oldAttr</var> is <var>attr</var>, return <var>attr</var>.
 
- <li><p>If <var>oldAttr</var> is non-null, then <a lt="replace an attribute">replace</a>
- <var>oldAttr</var> with <var>attr</var>.
+ <li>
+  <p>If <var>oldAttr</var> is non-null, then:</p>
+
+  <ul>
+   <li><p><a>Validate and set attribute value</a> <var>attr</var>'s <a for="Attr">value</a> for
+   <var>attr</var> with <var>element</var>.
+
+   <li><p>If <var>element</var> <a lt="has an attribute">has</a> an <a>attribute</a>
```

Comment:
I've read through this spec path and I believe that, if the default policy removes the existing attribute node, then this will call replace an attribute, which in turn calls replace a list item, which results in a no-op because the old value no longer exists to be replaced.

So I think in this situation you're probably right the spec doesn't need to do anything, will make that change.

Reply:
Having said that I'm slightly uneasy about stuff like attribute change steps still firing and how that all works spec wise.

Both Chromium and WebKit don't actually follow the spec 100% here and so I think would actually result in a different behaviour to this spec (the index lookup happens after the TT call) so that's also not ideal.